renderer/HardwareOpenGL.cpp

```diff
@@ -690,45 +690,46 @@ int opengl_Setup(oeApplication *app, int *width, int *height) {
   SDL_GL_SetAttribute(SDL_GL_BLUE_SIZE, 5);
   SDL_GL_SetAttribute(SDL_GL_DEPTH_SIZE, 16);
   SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
-  int flags = SDL_OPENGL;
+  Uint32 flags = SDL_WINDOW_OPENGL;
 
   if (fullscreen) {
-    flags |= SDL_FULLSCREEN;
+    flags |= SDL_WINDOW_FULLSCREEN;
   }
 
-  // Should we shoot for 32-bpp if available?  !!!
-  SDL_Surface *surface = SDL_SetVideoMode(*width, *height, 32, flags);
-  mprintf((0, "OpenGL: SDL GL surface is %sNULL.", (surface == NULL) ? "" : "NOT "));
 
-  if (!FindArg("-nomousecap")) {
-    // ShowCursor(0) and input grabbing need to be done before setting
-    //  the video mode, or the Voodoo 3 gets a hardware cursor stuck
-    //  on the screen.
-    // On modern systems this is not necessary and must be done after to take effect.
-    SDL_ShowCursor(0);
-    SDL_WM_GrabInput(SDL_GRAB_ON);
-    ddio_mouseGrabbed = true;
+  GSDLWindow = SDL_CreateWindow("Descent 3", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, *width, *height, flags);
+  if (!GSDLWindow) {
+    mprintf((0, "OpenGL: SDL window creation failed: %s", SDL_GetError()));
+    return 0;
   }
 
-  if (ddio_mouseGrabbed == false) {
-    SDL_WM_GrabInput(SDL_GRAB_OFF);
+  GSDLGLContext = SDL_GL_CreateContext(GSDLWindow);
+  if (!GSDLGLContext) {
+    mprintf((0, "OpenGL: OpenGL context creation failed: %s", SDL_GetError()));
+    SDL_DestroyWindow(GSDLWindow);
+    GSDLWindow = NULL;
+    return 0;
   }
 
-  SDL_WM_SetCaption("Descent 3", "Descent3");
+  if (!FindArg("-nomousecap")) {
+    ddio_mouseGrabbed = true;
+  }
+
+  SDL_SetRelativeMouseMode(ddio_mouseGrabbed ? SDL_TRUE : SDL_FALSE);
 
   // rcg09182000 gamma fun.
   // rcg01112000 --nogamma fun.
   if (FindArgChar("-nogamma", 'M')) {
     linux_permit_gamma = false;
   } else {
-    float f = Render_preferred_state.gamma;
-    bool gammarc = SDL_SetGamma(f, f, f);
-    linux_permit_gamma = (gammarc == 0);
+    Uint16 ramp[256];
+    SDL_CalculateGammaRamp(Render_preferred_state.gamma, ramp);
+    linux_permit_gamma = (SDL_SetWindowGammaRamp(GSDLWindow, ramp, ramp, ramp) == 0);
   } // else
 
   if (ParentApplication) {
     reinterpret_cast<oeLnxApplication *>(ParentApplication)->set_sizepos(0, 0, *width, *height);
-    SDL_FillRect(SDL_GetVideoSurface(), NULL, SDL_MapRGB(SDL_GetVideoSurface()->format, 255, 0, 0));
+    //SDL_FillRect(SDL_GetVideoSurface(), NULL, SDL_MapRGB(SDL_GetVideoSurface()->format, 255, 0, 0));
```

Comment:
Why did you choose to comment this line out instead of deleting it?

Reply:
My thinking was "in case this turned out to be important, we can revisit it later," but honestly it's probably not important, so I've removed the line in b362bc9c543f9688f6f269167f83c6606ae679aa.